crates/detect-targets/src/detect/linux.rs

```diff
@@ -93,22 +91,32 @@ You are not meant to run this directly.
     if status.success() {
         // Executing glibc ldd or /lib/ld-linux-{cpu_arch}.so.1 will always
         // succeeds.
-        String::from_utf8_lossy(&stdout)
-            .contains("GLIBC")
-            .then_some(Libc::Gnu)
+        let stdout = String::from_utf8_lossy(&stdout);
```

Comment:
The bug resides in here, from the log:

```
while loading shared libraries: --version: cannot open shared object file\n")
```

My guess is that on ubuntu-20.04, glibc isn't executable.

To fix this, we can simply check that one of these files exist instead of executing them.

Or we can also check os-release but that is too complex.